dataactvalidator/config/sqlrules/a32_appropriations.sql

```diff
@@ -0,0 +1,18 @@
+SELECT approp.row_number,
+	approp.allocation_transfer_agency,
+	approp.agency_identifier,
+	approp.beginning_period_of_availa,
+	approp.ending_period_of_availabil,
+	approp.availability_type_code,
+	approp.main_account_code,
+	approp.sub_account_code
+FROM appropriation AS approp
+WHERE approp.submission_id = {0}
+	AND EXISTS (
+		SELECT a.row_number,
+			a.tas
+		FROM appropriation AS a
+		WHERE a.row_number <> approp.row_number
+			AND a.tas IS NOT DISTINCT FROM approp.tas
+			AND a.submission_id = {0}
+	);
```

Comment:
Note that this'll display _two_ errors when there's a duplicate. If that's not okay (maybe we only want to display the second error or something), we can swap out 
`a.row_number <> approp.row_number` for `a.row_number > approp.row_number` or similar.

Super minor nit, but `a` may not be a good identifier here. I've seen a lot of "self" comparison functions use `other` or `right`